### DISCUSSION TITLE
UI/UXの確認と今後のアイデア

### DISCUSSION THREAD

**Discussion by kazzyfrog** [lgtm-factory/lgtm-factory#184] [Ideas]
現状、アイデアベースで、気になった点や、追加した方がいいかも？
などのアイデアをつらつら書いていきますー！

優先度の問題もあるので、あくまでアイデアですー

**Comment by kazzyfrog** [comment]
## ランダム生成機能について

個人的に、類似のすべてのサービスにあるので、ランダム機能欲しいなと思ってます！
サイトを訪れて、ボタン押しただけで、ワンクリックでコピーできるのは、やっぱり便利だなと思ってます！

実装するとしたら、以下のような感じになるかなと！
- 各デザインテーマに合う、おすすめのテキストや絵文字を複数用意しておく。
- ランダムボタンのクリックで、それらの中から「テーマ、それに紐付くおすすめの動的データ」が選択（コピー）される
- （確認のため、上記のランダムのデザインをモーダルで閲覧できる）

これに関しては、
デザインによって、文字数など、少しデザインに合わないケースなどもあるかと思います。
なので、あらかじめ見栄えの良いパラメータを用意しておくことで、推奨する使い方を提示できて、良いかなと思ってます！

**Reply by kazzyfrog** [reply]
下記にて、OGの問題もあるので、
めっちゃ簡素（β）でも、個別ページを作成するなら、
そこに、おすすめのテキストとかを、配置すればいいのかも。

デザインに紐づく、おすすめのパラメータをいくつか用意する。
例）
hot-coffee:
- Have a nice day!! :)
- Thank you !! Pull Request :)
- Thank You !! お疲れ様ー:)
- Have a Break 🍻 :)
- Awsomeeeeee!! ^w^

とか、変なところで、改行されていない、見栄えが良いパワメータをいくつか用意しておけば、
それを使ったり、そこから編集したりもできるかも！

１つのデザインで、複数パターン作れるので、ランダム機能になりそう！
10個デザインがあれば、50のの中からランダムとかでできそう！

**Reply by kazzyfrog** [reply]
１つのデザインに対し、複数パターンのパラメータを用意していくことで、
ランダム機能だけでなく、下記のように、フォームに表示するサジェスト機能を実装できるので、良さそう！！

https://qiita.com/waiwaiwait/items/1b5c8d48cc5d588d9ad3

**Comment by kazzyfrog** [comment]
## ヘッダーについて

利用規約やプライバシーポリシーなどは、一般的にはフッターにあります！
これは、直接的にいうと、ユーザーはほぼ見ない・見られなくて良いからかな、と思います！

今回のサービスにおいて、画像の利用方法の部分は、結構コアというか、重要度の高い部分だと思うので、
１回見てほしいページでもあるので、ヘッダーに置くのがいいのかなと、現状思ってます！

**Reply by kagomen** [reply]
なるほど！了解です！🫡

**Comment by kazzyfrog** [comment]
## Twitterシェアボタンについて

モーダルにある、Twitterシェアについて、
現状では、個別のデザイン画像を投稿するのではなく、WebサイトのトップページのURLでシェアしているかと思います！

理想としては、
デザインをクリックしたら表示されるシェアボタンなので、
そのデザインをシェアするボタンであることが、自然かなと思ってます！

ただ、URLを個別のデザインの、`/api/v1/lgtm-images?~~`のようなURLに設定しても、
TwitterでOG画像として表示されないと思います！（一応確認しました！）


これに関して、少しいい感じの実装を考えてみます👍

**Reply by kazzyfrog** [reply]
現状では、モーダルとは別で、
各デザインテーマの詳細ページを簡潔に作るという手が、最も簡単なのかなと思ってます！

んー、他には、
- Twitterでシェアするボタンなのに、OG画像は表示されないでも良いか
- サイトのトップページのシェアでも良いか

という点ですよね。。

**Reply by kazzyfrog** [reply]
というか、理想として、デザインと（できれば動的な内容まで）を、
OG画像でTwitterシェアできたら、めっちゃ良さそうだなと！実際見るのが１番興味が湧きそう。。

**Reply by kazzyfrog** [reply]
OG専用のために、ほぼ中身なくてもサクッと個別ページを作ってもいいのかも。
そうすれば、シェア問題は解決する！
各デザインで、シェアできるようになる

**Comment by kazzyfrog** [comment]
## モーダル内のUI/UXについて

下記のイシューに関するメモでもありますー
- #175

**Comment by kazzyfrog** [comment]
## 画像の読み込み・パフォーマンスについて


- 現状、トップページにアクセスするたびに、すべてのデザインのクエリが飛んで、毎回生成している状態なので、これキャッシュしたい
- あとは、エラー時の挙動・長めのテキストの時？にCPU処理時間が10msを超える問題も直したい

**Reply by kazzyfrog** [reply]
その他、画像の最適化は後！

**Reply by kazzyfrog** [reply]
デフォルトのテキストの、
TOPページに表示するようなデザインはキャッシュした方が良さそう。。！

**Reply by kazzyfrog** [reply]
その際、既存のAPIからfetchする？のか、
動的なクエリを受け取らないバージョンのほぼ同じAPIを静的に生やすのかは、
検証する必要がありそう！

多分、後者の方が楽で確実！

TOPページに表示する画像は、ビルド時に生成して画像として抱えた方がパフォーマンス向上！
